ONSdigital/sds
# SDSS-226 - List survey id query endpoint

### Motivation and Context

Create a new endpoint so that a list of all distinct survey id(s) can be provided/viewed.

### What has changed

- A new endpoint '/v1/survey_list' to query the schemas collection in Firestore and return the list of unique Survey IDs
- If there are no Survey IDs, then return 404 error with the appropriate error message
- Added unit and integration tests
- Regenerated openapi spec

### How to test?

- Post a new schema using the '/v1/schema' endpoint - this will create a new document in the 'schemas' collection in Firestore
- Create more such schemas in the Firestore collection using different Survey IDs
- Test the new endpoint '/v1/survey_list' - this should retrieve the unique Survey IDs from the 'schemas' collection
- Delete the 'schemas' collection in Firestore, the endpoint should then return the 404 error
- Pass unit and integration tests 

### Links

https://jira.ons.gov.uk/browse/SDSS-226
